src/discussions/post-comments/data/hooks.js

```diff
@@ -102,3 +104,80 @@ export function useCommentsCount(postId) {
 
   return commentsLength;
 }
+
+export const useDraftContent = () => {
+  const comments = useSelector(selectDraftComments);
+  const responses = useSelector(selectDraftResponses);
+
+  const getObjectByParentId = (data, parentId, isComment, id) => Object.values(data)
+    .find(draft => (isComment ? draft.parentId === parentId && (id ? draft.id === id : draft.isNewContent === true)
+      : draft.threadId === parentId && (id ? draft.id === id : draft.isNewContent === true)));
+
+  const updateDraftData = (draftData, newDraftObject) => ({
+    ...draftData,
+    [newDraftObject.id]: newDraftObject,
+  });
+
+  const addDraftContent = (content, parentId, id, threadId) => {
+    const data = parentId ? comments : responses;
+    const draftParentId = parentId || threadId;
+    const isComment = !!parentId;
+    const existingObj = getObjectByParentId(data, draftParentId, isComment, id);
+    const newObject = existingObj
+      ? { ...existingObj, content }
+      : {
+        threadId,
+        content,
+        parentId,
+        id: id || uuidv4(),
+        isNewContent: !id,
+      };
+
+    const updatedComments = parentId ? updateDraftData(comments, newObject) : comments;
+    const updatedResponses = !parentId ? updateDraftData(responses, newObject) : responses;
+
+    return { updatedComments, updatedResponses };
+  };
+
+  const getDraftContent = (parentId, threadId, id) => {
+    if (id) {
+      return parentId ? comments?.[id]?.content : responses?.[id]?.content;
+    }
+
+    const data = parentId ? comments : responses;
+    const draftParentId = parentId || threadId;
+    const isComment = !!parentId;
+
+    return getObjectByParentId(data, draftParentId, isComment, id)?.content;
+  };
+
+  const removeItem = (draftData, objId) => {
+    const { [objId]: _, ...newDraftData } = draftData;
+    return newDraftData;
+  };
+
+  const removeDraftContent = (parentId, id, threadId) => {
+    let updatedResponses = responses;
+    let updatedComments = comments;
+
+    if (!parentId) {
+      const responseObj = id ? responses[id] : getObjectByParentId(responses, threadId, false, id);
+      if (responseObj) {
+        updatedResponses = removeItem(responses, responseObj.id);
+      }
+    } else {
+      const commentObj = id ? comments[id] : getObjectByParentId(comments, parentId, true, id);
+      if (commentObj) {
+        updatedComments = removeItem(comments, commentObj.id);
+      }
+    }
```

Comment:
please consider a method for statements within if and else, the statements appear to be redundant
```
const commentObj = id ? comments[id] : getObjectByParentId(comments, parentId, true, id);
      if (commentObj) {
        updatedComments = removeItem(comments, commentObj.id);
      }
```
```
const responseObj = id ? responses[id] : getObjectByParentId(responses, threadId, false, id);
      if (responseObj) {
        updatedResponses = removeItem(responses, responseObj.id);
}